src/common/id/test/SegmentIdTest.cpp

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include <folly/concurrency/ConcurrentHashMap.h>
+#include <gflags/gflags.h>
+#include <gtest/gtest.h>
+
+#include "common/id/SegmentId.h"
+
+namespace nebula {
+class MockMetaClient : public meta::BaseMetaClient {
+ public:
+  folly::Future<StatusOr<int64_t>> getSegmentId(int64_t length) override {
+    std::lock_guard<std::mutex> guard(mutex_);
+    auto future = folly::makeFuture(cur_.load());
+    cur_.fetch_add(length);
+    return future;
+  }
+
+ private:
+  std::mutex mutex_;
+  std::atomic_int64_t cur_{0};
+};
+
+// In this case, small step cause that id run out before asyncFetchSegment() finish.
+// So one segment replace another one. (one is from asyncFetchSegment(), another if from
+// fetchSegment())
+TEST(SegmentIdTest, TestConcurrencySmallStep) {
```

Comment:
It's better to give a TestClass which could do following common initialization in SetUp functions for different test cases.